nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -0,0 +1,1211 @@
+                                                          self._ignored_scopes,
+                                                          self._target_scopes)
+        quant_prop_graph = self.set_allowed_quantization_types_for_operator_nodes(quant_prop_graph)
+        quant_prop_graph = self.setup_initial_quantizers(quant_prop_graph)
+
+        if self._run_consistency_checks:
+            quant_prop_graph.run_consistency_check()
+
+        iteration_counter = 0
+        while self._active_propagating_quantizers_queue:
+            if self._debug_interface is not None:
+                self._debug_interface.visualize_quantizer_propagation(self, quant_prop_graph, str(iteration_counter))
+            if self._run_consistency_checks:
+                quant_prop_graph.run_consistency_check()
+            prop_quantizer = self._active_propagating_quantizers_queue.pop()
+            quant_prop_graph = self.propagation_step(prop_quantizer, quant_prop_graph)
+            iteration_counter += 1
+
+        quant_prop_graph = self._filter_integer_input_quantizers(quant_prop_graph)
+
+        if self._debug_interface is not None:
+            self._debug_interface.visualize_quantizer_propagation(self, quant_prop_graph, "proposed")
+
+        if self._run_consistency_checks:
+            quant_prop_graph.run_consistency_check()
+
+        quantizer_setup = quant_prop_graph.create_quantizer_setup(self._weight_quantizable_node_names_vs_qconfigs)
+        insertions_vs_associated_prop_quants = self._map_quantization_points_to_prop_quantizers(
+            self._finished_propagating_quantizers, quant_prop_graph, quantizer_setup)
+
+        return QuantizationProposal(quantizer_setup=quantizer_setup,
+                                    quant_prop_graph=quant_prop_graph,
+                                    quantization_point_id_vs_prop_quantizer=insertions_vs_associated_prop_quants)
+
+    def _map_quantization_points_to_prop_quantizers(self,
+                                                    prop_quant_list: List[PropagatingQuantizer],
+                                                    quant_prop_graph: QuantizerPropagationStateGraph,
+                                                    quantizer_setup: MultiConfigQuantizerSetup) -> \
+            Dict[QuantizationPointId, PropagatingQuantizer]:
+        qps_vs_associated_prop_quants_dict = {}  # type: Dict[QuantizationPointId, PropagatingQuantizer]
+
+        for finished_prop_quantizer in prop_quant_list:
+            qip = quant_prop_graph.get_quant_insertion_point_for_propagating_quantizer(
+                finished_prop_quantizer)
+            for qp_id, qp in quantizer_setup.quantization_points.items():
+                if qp.insertion_point == qip:
+                    qps_vs_associated_prop_quants_dict[qp_id] = finished_prop_quantizer
+
+        return qps_vs_associated_prop_quants_dict
+
+    def get_final_quantizer_setup(self, finalized_quantization_proposal: FinalizedQuantizationProposal) -> \
```

Comment:
Could you please provide a description for the `finalized_quantization_proposal` parameter? Also, please add the `:return:` section.

Reply:
Done